server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java

```diff
@@ -75,7 +74,11 @@ public void restore(SnapshotFiles snapshotFiles, Store store, ActionListener<Voi
                 // this will throw an IOException if the store has no segments infos file. The
                 // store can still have existing files but they will be deleted just before being
                 // restored.
-                recoveryTargetMetadata = store.getMetadata(null, true);
+                if (isSearchableSnapshotStore(store.indexSettings().getSettings())) {
```

Comment:
We skip some of these steps here for searchable snapshots (as they are effectively useless, and require opening up the index, which we can defer to a later stage by special-casing this here).